src/lib/Behat/Page/ContentViewPage.php

```diff
@@ -150,6 +156,27 @@ public function addLocation(string $newLocationPath): void
         $this->universalDiscoveryWidget->confirm();
     }
 
+    public function addTranslation(string $language, string $base): void
+    {
+        $this->getHTMLPage()->find($this->getLocator('addTranslationButton'))->click();
+        $this->dialog->verifyIsLoaded();
+        $this->translationDialog->selectNewTranslation($language);
+        if ($base != 'none') {
+            $this->translationDialog->selectBaseTranslation($base);
+        }
+        $this->dialog->confirm();
+    }
```

Comment:
```suggestion
    public function addTranslation(string $language, string $base = 'none'): void
    {
        $this->getHTMLPage()->find($this->getLocator('addTranslationButton'))->click();
        $this->translationDialog->verifyIsLoaded();
        $this->translationDialog->selectNewTranslation($language);
        if ($base != 'none') {
            $this->translationDialog->selectBaseTranslation($base);
        }
        $this->translationDialog->confirm();
    }
```
TranslationDialog inherits from the Dialog class, so we can use the methods from the Dialog class - it executes the same code, but is more readable to me

Adding the default value could allow us to omit the second argument in the future (and improve DX a little bit).

Reply:
This will probably require adding these methods to the new component. I tried something like that initially and it didn't work OOTB.

Reply:
It should work out of the box, because TranslationDialog class extends the Dialog class.

Can you try, and if it doesn't work we will have a look together?

Reply:
It ends in InvalidArgumentException: `Could not find locator with identifier: 'confirm'. Available identifiers are: 'expandNewTranslationDropdown','expandBaseTranslationDropdown','dropdownItem'`.
It looks as if locators where not inherited and specifyLocators is a protected function.